### PR TITLE
refactor(web): Single theme entry point & ensure Mantine singletons

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?
* Move `ThemeProvider` to root of component tree. This ensures that everything has theming context available
* pin `@types/react` version via PNPM `overrides` to ensure that `@mantine/*` dependencies resolve to a singleton in the PNPM virtual filesystem
  * Pnpm v9 automatically installs optional deps by default when they match an existing semver, and we have other packages declaring `@types/react` as an optional dependency which results in multiple PNPM virtual fileystem paths for Mantine imports.
  * This resolves an issue where the color scheme couldn't be resolved due to lack of a `@mantine/core` singleton, which provides necessary context
* Use `AuthLayout` in routes to provide a single entry point for styling all unauthenticated pages.
* Declare `@mantine/core` as a `peerDependency` and `devDependency` in all packages except `@novu/web`

### Screenshots

https://github.com/novuhq/novu/assets/32132657/1d7ae862-e922-43dd-a131-b98a250a99df

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/web/src/AppRoutes.tsx [hunk 1]
```diff
@@ -53,28 +54,30 @@ export const AppRoutes = () => {
...
-      />
-      <Route
-        path={ROUTES.PARTNER_INTEGRATIONS_VERCEL_LINK_PROJECTS_EDIT}
-        element={
-          <EnsureOnboardingComplete>
-            <LinkVercelProjectPage type="edit" />
-          </EnsureOnboardingComplete>
-        }
-      />
+      <Route element={<AuthLayout />}>
```
Comment (rifont) [inline apps/web/src/AppRoutes.tsx]: Wrap all of these paths with `AuthLayout` with an `<Outlet/>` added to `AuthLayout` to render children. We now remove `AuthLayout` being duplicated in each of these child routes.

File: apps/web/src/Providers.tsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ const Providers: React.FC<PropsWithChildren<{}>> = ({ children }) => {
           </BrowserRouter>
         </QueryClientProvider>
       </SegmentProvider>
-    </ColorSchemeProvider>
+    </ThemeProvider>
```
Comment (rifont) [inline apps/web/src/Providers.tsx:60]: ThemeProvider now lives at the top of the tree.

File: apps/web/src/components/layout/components/AuthLayout.tsx [hunk 3]
```diff
@@ -1,52 +1,49 @@
...
+              display: 'none',
+            },
+          }}
+        >
           <img
-            src={CONTEXT_PATH + '/static/images/logo-formerly-dark-bg.png'}
+            src={CONTEXT_PATH + '/static/images/notifications/notification_01.png'}
             alt="logo"
-            style={{ alignSelf: 'flex-start', maxWidth: 150, marginTop: 5, marginLeft: 5 }}
+            style={{ maxWidth: 400 }}
```
Comment (rifont) [inline apps/web/src/components/layout/components/AuthLayout.tsx]: Bad diff, there's no changes here.

File: libs/novui/package.json [hunk 4]
```diff
@@ -80,7 +80,7 @@
     "@testing-library/react-hooks": "^8.0.1",
     "@testing-library/user-event": "^14.5.2",
     "@types/node": "^18.11.12",
-    "@types/react": "^17.0.1",
+    "@types/react": "^17.0.0",
```
Comment (rifont) [inline libs/novui/package.json:83]: Fix for consistency.

File: apps/web/src/AppRoutes.tsx (hunk 1 above)
Comment (SokratisVidros) [reply apps/web/src/AppRoutes.tsx]: How about we call the `AuthLayout`, `PublicPageLayout` to make it more clear as Auth layout doesn't tell us if this is for authenticated or unauthenticated context.

File: apps/web/src/Providers.tsx (hunk 2 above)
Comment (SokratisVidros) [reply apps/web/src/Providers.tsx:60]: 💯

File: apps/web/src/pages/auth/InvitationPage.tsx [hunk 5]
```diff
@@ -56,7 +55,7 @@ export default function InvitationPage() {
   }, [queryClient]);
 
   return (
-    <AuthLayout>
+    <>
```
Comment (SokratisVidros) [inline apps/web/src/pages/auth/InvitationPage.tsx]: This fragment can be removed. We can return AuthContainer or null.

File: apps/web/src/pages/auth/PasswordResetPage.tsx [hunk 6]
```diff
@@ -24,7 +22,7 @@ export function PasswordResetPage({}: Props) {
   }
 
   return (
-    <AuthLayout>
+    <>
```
Comment (SokratisVidros) [inline apps/web/src/pages/auth/PasswordResetPage.tsx]: Ditto

**Review by SokratisVidros — APPROVED** [review]
💯

File: apps/web/src/AppRoutes.tsx (hunk 1 above)
Comment (rifont) [reply apps/web/src/AppRoutes.tsx]: Great suggestion. I'll change this.

**Pull request merged**